streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -509,4 +556,33 @@ public void shouldDeleteCheckpointFileOnCreationIfEosEnabled() throws Exception
         }
     }
 
+    private ProcessorStateManager getStandByStateManager(TaskId taskId) throws IOException {
+        return new ProcessorStateManager(
+            taskId,
+            noPartitions,
+            true,
+            stateDirectory,
+            new HashMap<String, String>() {
+                {
+                    put(persistentStoreName, persistentStoreTopicName);
+                }
+            },
+            changelogReader,
+            false);
+    }
+
+    private MockStateStoreSupplier.MockStateStore getPersistentStore() {
+        return new MockStateStoreSupplier.MockStateStore("persistentStore", true);
+    }
+
+    private static class MockBatchingRestoreCallback extends AbstractNotifyingBatchingRestoreCallback {
```

Comment:
We have this in `CompositeRestoreListenerTest` as well, also there is a similar `MockNoListenBatchingStateRestoreCallback` in that class as well. Could we collapse them all into a single class (we already have a `MockStateRestoreListener` in `o.a.k.test`, probably can just extend to another class under `o.a.k.test`)?